pkg/knowledgebase/path_satisfaction.go

```diff
@@ -9,14 +10,16 @@ import (
 
 type (
 	PathSatisfaction struct {
-		AsTarget []PathSatisfactionRoute `json:"as_target" yaml:"as_target"`
-		AsSource []PathSatisfactionRoute `json:"as_source" yaml:"as_source"`
+		AsTarget            []PathSatisfactionRoute `json:"as_target" yaml:"as_target"`
+		AsSource            []PathSatisfactionRoute `json:"as_source" yaml:"as_source"`
+		DenyClassifications []string                `yaml:"deny_classifications"`
```

Comment:
I'm not sure I understand what `DenyClassifications` is used for / what problems it solves. Could you document and add a test case (either in path_selection or as a functional test) demonstrating it its use-case?

Reply:
was just carry over from the other stuff so can remove. But it was a way of making sure subnets arent used in permissions path solving and iam roles arent used in network path solving since they were getting connected by the ec2 launch templates security group. Since this isnt tackling dynamic port mappings anymore i can remove it

Reply:
Up to you, but the choices IMO are: (1) keep it in and add tests/docs for it, (2) split it out (either to be reimplemented later or keep around as an incomplete PR)